src/data_types/w3c/credential_proof.rs

```diff
@@ -0,0 +1,113 @@
+use crate::data_types::w3c::presentation_proof::CredentialPresentationProof;
+use crate::utils::encoded_object::EncodedObject;
+use anoncreds_clsignatures::{
+    CredentialSignature as CLCredentialSignature, RevocationRegistry, SignatureCorrectnessProof,
+    Witness,
+};
+use serde_json::Value;
+
+#[allow(clippy::large_enum_variant)]
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum CredentialProof {
+    AnonCredsSignatureProof(CredentialSignatureProof),
+    AnonCredsCredentialPresentationProof(CredentialPresentationProof),
+    NonAnonCredsDataIntegrityProof(NonAnonCredsDataIntegrityProof),
+}
+
+#[derive(Debug, Clone, Default, PartialEq, Eq, Serialize, Deserialize)]
+pub struct CredentialSignatureProof {
+    #[serde(rename = "type")]
+    pub type_: CredentialSignatureType,
+    pub signature: String,
+}
+
+impl CredentialSignatureProof {
+    pub fn new(signature: CredentialSignature) -> Self {
+        CredentialSignatureProof {
+            type_: CredentialSignatureType::AnonCredsProof2023,
+            signature: signature.encode(),
+        }
+    }
+
+    pub fn get_credential_signature(&self) -> crate::Result<CredentialSignature> {
+        match self.type_ {
+            CredentialSignatureType::AnonCredsProof2023 => {
+                CredentialSignature::decode(&self.signature)
+            }
+        }
+    }
+}
+
+pub type NonAnonCredsDataIntegrityProof = Value;
```

Comment:
Very nitty, but I prefer to keep it with the name space for `Value` as it is quite ambigous. `serde_json::Value` makes it easier to read.